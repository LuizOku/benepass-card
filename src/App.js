import React, { useState, useCallback } from "react";
import { Card, Checkbox } from "./components";

const MOCK_CARD = {
  number: "1232222344321732",
  type: "virtual",
  validThru: "08/28",
  cvc: "345",
  zip: "66062",
};

function App() {
  const [shouldShowDetails, setSetshouldShowDetails] = useState(false);

  const handleCheckboxChange = useCallback(() => {
    setSetshouldShowDetails(!shouldShowDetails);
  }, [shouldShowDetails]);

  return (
    <div className="w-full min-h-screen p-6 sm:py-[60px] sm:px-[55px] flex flex-col">
      <h1 className="text-[26px] leading-8 font-medium">Flex Card</h1>
      <span className="text-xl leading-8 font-normal mb-8">
        Used for pre-tax purchases
      </span>
      <Card {...MOCK_CARD} shouldShowDetails={shouldShowDetails} />
      <Checkbox
        className="mt-4"
        label="Show details"
        value={shouldShowDetails}
        onChange={handleCheckboxChange}
      />
    </div>
  );
}

export default App;
