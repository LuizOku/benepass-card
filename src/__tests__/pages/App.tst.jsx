// Import necessary libraries and components
import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../../App";

const renderComponent = () => {
  return render(<App />);
};

describe("<App />", () => {
  it("renders without crashing", () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it("should render the page title", () => {
    renderComponent();
    expect(screen.getByText("Flex Card")).toBeInTheDocument();
  });

  it("should render the page subtitle", () => {
    renderComponent();
    expect(screen.getByText("Used for pre-tax purchases")).toBeInTheDocument();
  });

  it("should render the Card component", () => {
    renderComponent();
    expect(screen.getByTestId("card-component")).toBeInTheDocument();
  });

  it("should render the Checkbox component", () => {
    renderComponent();
    expect(screen.getByTestId("checkbox-component")).toBeInTheDocument();
  });
});
