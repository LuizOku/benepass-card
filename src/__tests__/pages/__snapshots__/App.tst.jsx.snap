// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full min-h-screen p-6 sm:py-[60px] sm:px-[55px] flex flex-col"
      >
        <h1
          class="text-[26px] leading-8 font-medium"
        >
          Flex Card
        </h1>
        <span
          class="text-xl leading-8 font-normal mb-8"
        >
          Used for pre-tax purchases
        </span>
        <div
          class="bg-card bg-cover rounded-xl p-6 sm:w-[392px] h-[272px] min-w-[310px] flex flex-col transition-all duration-500 md:hover:scale-105"
          data-testid="card-component"
        >
          <div
            class="w-full flex items-center justify-between mb-6"
          >
            <img
              alt="Logo"
              class="w-[121px] h-8 object-cover"
              data-testid="card-logo"
              src="Logo.svg"
            />
            <span
              class="px-3 py-1 rounded-full bg-white capitalize text-base font-medium text-brand-indigo-100"
            >
              virtual
            </span>
          </div>
          <span
            class="text-[26px] leading-8 text-white"
          >
            ●●●● 1732
          </span>
          <div
            class="mt-6 flex flex-row items-center justify-between w-1/2"
          >
            <div
              class="flex flex-col"
            >
              <span
                class="text-xs font-medium text-white"
              >
                VALID THRU
              </span>
              <span
                class="text-[18px] leading-6 font-medium text-white"
              >
                08/28
              </span>
            </div>
            <div
              class="flex flex-col"
            >
              <span
                class="text-xs font-medium text-white"
              >
                CVC
              </span>
              <span
                class="text-[18px] leading-6 font-medium text-white"
              >
                ●●●
              </span>
            </div>
          </div>
          <div
            class="flex-grow"
          />
          <div
            class="w-full flex items-center justify-between"
          >
            <span
              class="text-[18px] leading-6 font-normal text-white"
            >
              ZIP 66062
            </span>
            <img
              alt="Visa"
              class="w-[54px] h-[18px] object-cover"
              data-testid="visa-img"
              src="Visa.svg"
            />
          </div>
        </div>
        <div
          class="flex items-center mt-4"
          data-testid="checkbox-component"
        >
          <input
            class="mr-3 transform scale-110 border-0 outline accent-brand-pink-100 cursor-pointer transition-all duration-500 md:hover:scale-125"
            data-testid="checkbox"
            type="checkbox"
          />
          <label
            class="text-base font-normal false"
          >
            Show details
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full min-h-screen p-6 sm:py-[60px] sm:px-[55px] flex flex-col"
    >
      <h1
        class="text-[26px] leading-8 font-medium"
      >
        Flex Card
      </h1>
      <span
        class="text-xl leading-8 font-normal mb-8"
      >
        Used for pre-tax purchases
      </span>
      <div
        class="bg-card bg-cover rounded-xl p-6 sm:w-[392px] h-[272px] min-w-[310px] flex flex-col transition-all duration-500 md:hover:scale-105"
        data-testid="card-component"
      >
        <div
          class="w-full flex items-center justify-between mb-6"
        >
          <img
            alt="Logo"
            class="w-[121px] h-8 object-cover"
            data-testid="card-logo"
            src="Logo.svg"
          />
          <span
            class="px-3 py-1 rounded-full bg-white capitalize text-base font-medium text-brand-indigo-100"
          >
            virtual
          </span>
        </div>
        <span
          class="text-[26px] leading-8 text-white"
        >
          ●●●● 1732
        </span>
        <div
          class="mt-6 flex flex-row items-center justify-between w-1/2"
        >
          <div
            class="flex flex-col"
          >
            <span
              class="text-xs font-medium text-white"
            >
              VALID THRU
            </span>
            <span
              class="text-[18px] leading-6 font-medium text-white"
            >
              08/28
            </span>
          </div>
          <div
            class="flex flex-col"
          >
            <span
              class="text-xs font-medium text-white"
            >
              CVC
            </span>
            <span
              class="text-[18px] leading-6 font-medium text-white"
            >
              ●●●
            </span>
          </div>
        </div>
        <div
          class="flex-grow"
        />
        <div
          class="w-full flex items-center justify-between"
        >
          <span
            class="text-[18px] leading-6 font-normal text-white"
          >
            ZIP 66062
          </span>
          <img
            alt="Visa"
            class="w-[54px] h-[18px] object-cover"
            data-testid="visa-img"
            src="Visa.svg"
          />
        </div>
      </div>
      <div
        class="flex items-center mt-4"
        data-testid="checkbox-component"
      >
        <input
          class="mr-3 transform scale-110 border-0 outline accent-brand-pink-100 cursor-pointer transition-all duration-500 md:hover:scale-125"
          data-testid="checkbox"
          type="checkbox"
        />
        <label
          class="text-base font-normal false"
        >
          Show details
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
